### ISSUE TITLE
Parallel Execution of Processors in Cluster Mode

### ISSUE THREAD

**Issue by Tianjian** [WeBankFinTech/FATE#2]
**Is your feature request related to a problem? Please describe.**
All Processors of Eggroll are running in  one python process, pretty slow

**Describe the solution you'd like**
Since GIL problem(Feature?) of PVM can not be fixed, deploying multiple processor would be a easy solution

**Describe alternatives you've considered**
Jython

**Comment by maxwong** [comment]
Thanks for your feedback, Tianjian. This is a known issue and we are looking into it at high priority. 
Eden and I have been discussing it. And we have come up with 2 approaches:

**Approach 1: Using a process pool (instead of thread pool) to further parallelize tasks**
We investigated this approach even before our initial release. In one of our internal alpha version we tested an implementation by Brian Quinlan in standalone version. It worked not bad except for two issues:
a. Callback results are sometimes missing. But this has been resolved by Eden;
b. Lambda function is not supported (due to pickle mechanism used by this implementation). But we expect lambda functions will be intensively used by users.

As a result, process pool is not used in our initial release. We have to fix issue b above to use this approach.

**Approach 2: Implementing a node-manager-alike component which manages processes and runtime environments**
Yarn / Kubernetes has similar mechanisms (with different names). This approach provides user more fine-grained mechanisms to manage a node. Besides, we can evolve to a Yarn/K8S-compatible infrastructure so that users can make use of their existing tools.

These 2 approaches do not conflict with each other. We are discussing it and will update once we have any progress.
